### PR TITLE
Use Math.hypot + various fixes

### PR DESCRIPTION
**Pull request by chicoxyzzy**
All the changes:
- Use Math.hypot
- Fix context parameter name in isTextElement function
- Remove unused code
- Use block scope everywhere

Since ES2015 feature `Math.imul` is used in a code base already, it should be fine to use `Math.hypot` also.

Should the `target` in `tsconfig` be `"es2015"`?

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
Line break change in `package.json` is made by prettier

File: src/index.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ type ExcalidrawTextElement = ExcalidrawElement & {
 const LOCAL_STORAGE_KEY = "excalidraw";
 const LOCAL_STORAGE_KEY_STATE = "excalidraw-state";
 
-var elements = Array.of<ExcalidrawElement>();
+let elements = Array.of<ExcalidrawElement>();
```
Comment (vjeux) [inline src/index.tsx]: Since you're changing it, might as well make it const :)

File: src/index.tsx [hunk 2]
```diff
@@ -69,7 +69,7 @@ function distanceBetweenPointAndSegment(
 
   const dx = x - xx;
   const dy = y - yy;
-  return Math.sqrt(dx * dx + dy * dy);
+  return Math.hypot(dx, dy);
```
Comment (vjeux) [inline src/index.tsx]: Mind blown, no idea that it was a thing!

File: src/index.tsx [hunk 3]
```diff
@@ -233,18 +233,6 @@ function renderScene(
...
-  let minY = Infinity;
-  let maxY = 0;
-
-  elements.forEach(element => {
-    minX = Math.min(minX, getElementAbsoluteX1(element));
-    maxX = Math.max(maxX, getElementAbsoluteX2(element));
-    minY = Math.min(minY, getElementAbsoluteY1(element));
-    maxY = Math.max(maxY, getElementAbsoluteY2(element));
-  });
-
```
Comment (vjeux) [inline src/index.tsx]: Why did you delete all of this, is it unused?

File: src/index.tsx [hunk 4]
```diff
@@ -435,7 +423,7 @@ function generateDraw(element: ExcalidrawElement) {
         { stroke: element.strokeColor, fill: element.backgroundColor }
       )
     );
-    element.draw = (rc, contex, { scrollX, scrollY }) => {
```
Comment (vjeux) [inline src/index.tsx]: how did it even work!?

File: src/index.tsx (hunk 1 above)
Comment (chicoxyzzy) [reply src/index.tsx]: `elements` are reassigned in the `restore` function

File: src/index.tsx (hunk 3 above)
Comment (chicoxyzzy) [reply src/index.tsx]: Yes, there are no `minX`, `maxX`, `minY`, `maxY` usages elsewhere

File: src/index.tsx (hunk 1 above)
Comment (vjeux) [reply src/index.tsx]: arg, would you mind rewriting the assignment in restore to mutate the array in place?

```js
elements.splice(0, elements.length, ...newElements);
```

This way `const` would work and prevent mistakes where people accidentally capture it and it gets stale.

**Comment by vjeux** [comment]
Actually, let me merge this, we can do the elements changes in another PR
